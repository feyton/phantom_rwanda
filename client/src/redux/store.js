import { configureStore } from '@reduxjs/toolkit';
import { authReducer } from './reducers/authReducer.js';
import { busMovementReducer } from './reducers/busMovementReducer.js';
import { counterReducer } from './reducers/counterReducer.js';
import { dashboardReducer } from './reducers/dashboardReducer.js';
import { managementReducer } from './reducers/managementReducer.js';
import { profileReducer } from './reducers/profileReducer.js';
import { simulateReducer } from './reducers/simulateReducer.js';
import { socketReducer } from './reducers/socketReducer.js';
import { updateProfileReducer } from './reducers/updateProfileReducer.js';

export const store = configureStore({
  reducer: {
    counter: counterReducer,
    profileUpdates: updateProfileReducer,
    profileInfo: profileReducer,
    auth: authReducer,
    management: managementReducer,
    dashboard: dashboardReducer,
    socket: socketReducer,
    simulate: simulateReducer,
    movement: busMovementReducer
  }
});
