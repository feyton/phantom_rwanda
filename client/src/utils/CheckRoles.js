import React from 'react';
import { useSelector } from 'react-redux';
import { Link } from 'react-router-dom';

function CheckRole({ children, role, type }) {
  const { authenticated, user } = useSelector((state) => state?.auth);
  if (!authenticated) {
    return '';
  }
  if (user.role === 'admin') {
    return children;
  }
  if (role.includes(user.role)) {
    /* istanbul ignore next */
    return children;
  }
  if (type === 'page' && !role.includes(user.role)) {
    return (
      <div className="flex flex-col items-center justify-center">
        <h3 className="font-rale font-bold text-3xl mb-4 mt-5">
          You don't have the permissions to view the page.
        </h3>
        <Link
          to="/login"
          className="bg-primary rounded-sm font-bold py-1 px-3 text-white"
        >
          Login with a different account
        </Link>
      </div>
    );
  }
  return '';
}

export default CheckRole;
