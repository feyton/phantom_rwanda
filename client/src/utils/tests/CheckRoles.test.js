import React from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { store } from '../../redux/store.js';
import CheckRoles from '../CheckRoles.js';

describe('CheckRoles component', () => {
  it('should render the ManageDropdownRoute component', () => {
    const elem = renderer
      .create(
        <Provider store={store}>
          <MemoryRouter>
            <CheckRoles />
          </MemoryRouter>
        </Provider>
      )
      .toJSON();
    expect(elem).toMatchSnapshot();
  });
});
