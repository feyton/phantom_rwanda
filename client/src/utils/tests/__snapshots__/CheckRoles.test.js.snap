// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckRoles component should render the ManageDropdownRoute component 1`] = `""`;
